web/src/main/java/ua/com/vertex/logic/LoggingLogicImpl.java

```diff
@@ -67,19 +67,21 @@ public String setUser(String email, Model model) throws Exception {
     }
 
     private String requiredView() throws Exception {
-        String view;
+        String view = null;
 
-        List<String> authorities = new ArrayList<>();
-        SecurityContextHolder.getContext().getAuthentication().getAuthorities()
-                .forEach(e -> authorities.add(e.toString()));
+        Collection authorities = SecurityContextHolder.getContext().getAuthentication().getAuthorities();
+        Iterator iterator = authorities.iterator();
 
         if (authorities.size() == 1) {
-            if (ADMIN.equals(authorities.get(0))) {
-                view = ADMIN_PAGE;
-            } else if (USER.equals(authorities.get(0))) {
-                view = USER_PAGE;
-            } else {
-                throw new Exception(AUTHORITIES_ERROR);
+            while (iterator.hasNext()) {
+                String result = iterator.next().toString();
+                if (ADMIN.equals(result)) {
+                    view = ADMIN_PAGE;
+                } else if (USER.equals(result)) {
+                    view = USER_PAGE;
+                } else {
+                    throw new Exception(AUTHORITIES_ERROR);
+                }
```

Comment:
because 1. you use iterator when it is not needed, 2. only last result will form returning result, but exception can be thrown in any iteration of the loop 3. map with ADMIN - ADMIN_PAGE as key-value can be used to get correct page.